ld-pruning-wf.wdl

```diff
@@ -0,0 +1,441 @@
+			The CWL uses nameroot for this, but I think the WDL needs the full path
+			Ex: "inputs/test_data_chrX.vcf.gz" returns ["inputs/test_data_", "X"]
+			'''
+			py_split = py_filename.split("chr")
+			if unicode(str(py_split[1][1])).isnumeric():
+				# chr10 and above
+				py_thisNamerootSplit = py_split[0]
+				py_thisChr = py_split[1][0:2]
+			else:
+				# chr9 and below + chrX
+				py_thisNamerootSplit = py_split[0]
+				py_thisChr = py_split[1][0:1]
+			return [py_thisNamerootSplit, py_thisChr]
+
+		gds = "~{gds_n_varinc.left}"
+		variant_include_file = "~{gds_n_varinc.right}"
+
+		f = open("subset_gds.config", "a")
+		f.write("gds_file " + gds + "\n")
+		f.write("variant_include_file " + variant_include_file + "\n")
+
+		if "~{defSampleInclude}" == "true":
+			f.write("sample_include_file ~{sample_include_file}\n")
+
+		if ("~{out_prefix}" != ""):
+			chromosome = py_rootPlusChr(gds)[1]
+			py_filename = "~{out_prefix}" + "_chr" + chromosome + ".gds"
+		else:
+			chromosome = py_rootPlusChr(gds)[1]
+			basename = os.path.basename(py_rootPlusChr(gds)[0])
+			py_filename = basename + "subset_chr" + chromosome + ".gds"
+
+		f.write("subset_gds_file " + py_filename)
+		f.close()
+		CODE
+
+		R -q --vanilla < /usr/local/analysis_pipeline/R/subset_gds.R --args subset_gds.config
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master:2.10.0"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File config_file = "subset_gds.config"
+		File subset_output = glob("*.gds")[0]
+	}
+}
+
+# [3] merge_gds -- merge a bunch of GDS files into one
+task merge_gds {
+	input {
+		Array[File] gdss
+		String? out_prefix
+
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int gds_size = ceil(size(gdss, "GB"))
+	Int final_disk_dize = gds_size * 3 + addldisk
+	String filename = select_first([out_prefix, "merged"])
+
+	command <<<
+		set -eux -o pipefail
+
+		# CWL has an ln -s, will probably need to use copy trick again
+		echo "Copying inputs into the workdir"
+		BASH_FILES=(~{sep=" " gdss})
+		for BASH_FILE in ${BASH_FILES[@]};
+		do
+			cp ${BASH_FILE} .
+		done
+
+		echo "Generating config file"
+		python << CODE
+		import os
+
+		def find_chromosome(file):
+			chr_array = []
+			chrom_num = split_on_chromosome(file)
+			if(unicode(str(chrom_num[1][1])).isnumeric()):
+				# two digit number
+				chr_array.append(chrom_num[1][0])
+				chr_array.append(chrom_num[1][1])
+			else:
+				# one digit number or Y/X/M
+				chr_array.append(chrom_num[1][0])
+			return "".join(chr_array)
+
+		def split_on_chromosome(file):
+			# if input is "amishchr1.gds"
+			# output is ["amish", ".gds", "chr"]
+			chrom_num = file
+			if "chr" in chrom_num:
+				chrom_num = chrom_num.split("chr")
+				chrom_num.append("chr")
+			else:
+				return "error-invalid-inputs"
```

Comment:
That is the intended behavior.

find_chromosome() is only called once, and it will correctly recognize if that passed in string is valid or not. This same logic is used in vcf-to-gds and I think we've discussed it before. Since it seems to have come up twice, maybe it would be clearer to have it exit within split_on_chromosome() instead?

```
	this_chr = find_chromosome(gds_file)
	if this_chr == "error-invalid-inputs":
		print("Unable to determine chromosome number from inputs.")
		print("Please ensure your files contain ''chr'' followed by")
		print("the number of letter of the chromosome (chr1, chr2, etc)")
		exit(1)
	else:
		chr_array.append(this_chr)
```

Reply:
I don't love this check here, and I didn't like it in the vcf-to-gds wdl either, as you know. By adding it, we are adding a check that the CWL does not have, and that increases the code base to maintain. (It's not a bad check to add, if we are adding stuff, though.)

As for how the split_on_chromosome() should behave, it strikes me as odd throw an error by passing a string to a second function (find_chromosom()) that then ignores it through odd logic, then the original string which is acting like a pseudo-error is passed to main() which THEN recognizes it as the pseudo-error that it is. It would be simpler to just raise an exception in split_on_chromosome(), or you could just exit() in that function.